### ISSUE TITLE
infinite loop in wikibooks?

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#479]
I have just been rebuilding 2019-03-01 enbooks and have found that this does not seem to finish
That is, I have to crash out of the process to stop

On further investigation, I think I have found (at least) two pages that tie up xowa-gui
Category:Book:Character_List_for_Karlgren's_GSR
Category:Book:Character_List_for_Baxter&Sagart

These pages end up calling Module:Wikilisp and never terminating - but I cannot work out why

**Comment by gnosygnu** [comment]
Yeah, this looks like a doozy. I spent some time on it tonight, but couldn't get further than en.wikibooks.org/wiki/Template:BookCat/category. I think the problem is related to this ```{{BookCat/category/subjects|book={{BOOKNAME|{{{fullpagename}}}}}}}{{BookCat/category/shelves|book={{BOOKNAME|{{{fullpagename}}}}}}}``` but am not sure how. Offhand it looks like expandTemplate is being called repeatedly. I suspect some sort of endless ```lst``` parser function call, but haven't been able to find

Will try to return back to it for later debugging.
